astropy/io/ascii/ui.py

```diff
@@ -271,6 +275,16 @@ def _guess(table, read_kwargs, format, fast_reader):
             # user supplies delimiter="|" but the guess wants to try delimiter=" ",
             # so skip the guess entirely.
             continue
+
+        if guess_kwargs not in filtered_guess_kwargs:
+            filtered_guess_kwargs.append(guess_kwargs)
+
+    # If there are not at least two formats to guess then return no table and
+    # indicate that guessing did not occur.
+    if len(filtered_guess_kwargs) <= 1:
+        return None, False  # dat, guess
```

Comment:
I see why you did it and I think itś a clever way to do it, but its logic might need a comment or two to explain.